src/storage/src/table/storage_table/iter_utils.rs

```diff
@@ -81,6 +84,71 @@ where
     }
 }
 
+/// Merge two streams of primary key and rows into a single stream, sorted by primary key.
+/// We should ensure that the primary key from different streams are unique.
+#[try_stream(ok = (Cow<'a, Row>, Cow<'a, Row>), error = StorageError)]
+pub async fn merge_by_pk<'a, S>(
+    stream1: S,
+    pk_indices1: &'a [usize],
+    stream2: S,
+    pk_indices2: &'a [usize],
```

Comment:
The lifetime annotation for pk_indices seems unnecessary. But removing those won't compile.